File: Share/src/main/java/com/buguagaoshu/share/domain/ViewCountLog.java
package com.buguagaoshu.share.domain;

import jakarta.persistence.*;
import lombok.Data;

/**
 * @author Pu Zhiwei {@literal puzhiweipuzhiwei@foxmail.com}
 * @create 2024-05-24
 * 文件分享历史记录查看
 */
@Data
@Entity
@Table(name = "view_table")
public class ViewCountLog {
    @jakarta.persistence.Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    /**
     * 0 share
     * 1 file
     * */
    private Integer type;

    private String ip;

    private String city;

    private String ua;

    /**
     * 文件或分享文本的ID
     * */
    private Long targetId;

    public Long viewTime;

}

File: Go/internal/model/view_count_log.go
package model

// ViewCountLog 对齐 Spring com.buguagaoshu.share.domain.ViewCountLog（表 view_table）
type ViewCountLog struct {
	ID       *int64 `gorm:"primaryKey;autoIncrement;column:id" json:"id"`
	// 0 share, 1 file, 2 ai
	Type     *int   `gorm:"column:type" json:"type"`
	Ip       string `gorm:"column:ip" json:"ip"`
	City     string `gorm:"column:city" json:"city"`
	Ua       string `gorm:"column:ua" json:"ua"`
	TargetId *int64 `gorm:"column:target_id" json:"targetId"`
	ViewTime *int64 `gorm:"column:view_time" json:"viewTime"`
}

func (ViewCountLog) TableName() string { return "view_table" }
